apps/wallet-mobile/src/yoroi-wallets/hooks/index.ts

```diff
@@ -985,34 +994,83 @@ export const useNativeAssetImage = ({
   width,
   height,
   mediaType = 'image/webp',
-  resizeMode = 'cover',
+  contentFit = 'cover',
   kind = 'metadata',
+  responseType = 'binary',
 }: NativeAssetImageRequest) => {
   const network = networkId === 300 ? 'preprod' : 'mainnet'
   const pWidth = PixelRatio.getPixelSizeForLayoutSize(Number(width))
   const pHeight = PixelRatio.getPixelSizeForLayoutSize(Number(height))
-  const isMediaTypeSupported = supportedTypes.includes(mediaType.toLocaleLowerCase())
+  const lcMediaType = mediaType.toLocaleLowerCase()
+  const isMediaTypeSupported = supportedTypes.includes(lcMediaType)
+  const needsGif = lcMediaType === 'image/gif' && Platform.OS === 'ios'
+  const mimeType = needsGif ? 'image/gif' : 'image/webp'
+  const headers = useMemo(
+    () => ({
+      Accept: responseType === 'binary' ? mimeType : 'text/plain',
+      'X-Encoded-Mimetype': mimeType,
+    }),
+    [mimeType, responseType],
+  )
+  const queryClient = useQueryClient()
+
+  const [isError, setError] = React.useState(false)
+  const [isLoading, setLoading] = React.useState(false)
+
+  const queryKey = ['native-asset-img', policy, name, responseType, `${pWidth}x${pHeight}`, contentFit]
+
   const query = useQuery({
+    enabled: isMediaTypeSupported,
     staleTime: Infinity,
-    queryKey: ['native-asset-img', policy, name, `${width}x${height}`, resizeMode],
-    queryFn: async () => {
-      const response = await fetch(
-        `https://${network}.cardano-nativeassets-prod.emurgornd.com/${policy}/${name}?width=${pWidth}&height=${pHeight}&kind=${kind}&fit=${resizeMode}`,
-      )
+    queryKey,
+    queryFn: async (context) => {
+      const count = queryClient.getQueryState(context.queryKey)?.dataUpdateCount
+      const cache = count ? `&cache=${count}` : ''
+      const requestUrl = `https://${network}.cardano-nativeassets-prod.emurgornd.com/${policy}/${name}?width=${pWidth}&height=${pHeight}&kind=${kind}&fit=${contentFit}${cache}`
+
+      if (responseType === 'binary') {
+        setLoading(true)
+        setError(false)
+        return requestUrl
+      }
+
+      const response = await fetch(requestUrl, {
+        headers,
+      })
       if (!response.ok) {
-        throw new Error(`NativeAsset CDN response was not ok for policy=${policy} name=${name}`)
+        throw new Error(`NativeAsset CDN: Not ok - ${requestUrl}`)
       }
       if (response.status === 201 || response.status === 202) {
-        throw new Error(`NativeAsset CDN still processing policy=${policy} name=${name}`)
+        throw new Error(`NativeAsset CDN: Processing - ${requestUrl}`)
       }
-      return `data:image/webp;base64,${await response.text()}`
+      return `data:${mimeType};base64,${await response.text()}`
     },
-    enabled: isMediaTypeSupported,
   })
 
+  const timerRef = React.useRef<any>(0)
+  React.useEffect(() => () => clearTimeout(timerRef.current), [])
```

Comment:
it supposed to work.
```suggestion
  const timerRef = React.useRef<number | null>(null)
  React.useEffect(() => () => clearTimeout(timerRef.current), [])
```

Reply:
It complains because `node.js` types are interfering with react-native's https://dev.to/retyui/react-native-fixing-typescript-issue-type-timeout-is-not-assignable-to-type-number-10hg

`number|null` doesn't work. I can try to find a fully compliant solution if `any` is too bad

Reply:
This one seems to silence the error
```
  const timerRef = React.useRef<ReturnType<typeof setTimeout>>()
```

Reply:
👀  https://github.com/facebook/react-native/blob/cd6755f2720dd6787545834ffd83905fd2d7517b/packages/react-native/types/modules/globals.d.ts#L27

Reply:
I'm guessing is the `null` issue.